lib/gilded_rose.ex

```diff
@@ -7,77 +7,160 @@ defmodule GildedRose do
+    |> different_aged_backstage()
+    |> diff_sulfuras_sellin()
+    |> sell_in_less_zero()
+  end
+
+  # Helper methods for Conjure functionality
+  defp degrade_quality(item, rate) do
+    %{item | quality: max(item.quality - rate, 0)}
+  end
+
+  defp reduce_sell_in_by_one(item) do
+    %{item | sell_in: item.sell_in - 1}
+  end
+
+  # Helper methods Refactor
+  defp sell_in_less_zero(%Item{sell_in: sell_in} = item) when sell_in < 0 do
+    item
+    |> diff_aged()
+  end
+
+  defp sell_in_less_zero(item) do
+    item
+  end
+
+  defp diff_aged(%Item{name: name} = item) when name != "Aged Brie" do
+    item
+    |> diff_backstate()
+  end
+
+  defp diff_aged(item) do
+    item
+    |> max_quality()
+  end
+
+  defp increase_quality(item) do
+    %{item | quality: item.quality + 1}
+  end
+
+  defp max_quality(%Item{quality: quality} = item) when quality < 50 do
+    item
+    |> increase_quality()
+  end
+
+  defp max_quality(item) do
+    item
+  end
+
+  defp diff_backstate(%Item{name: name} = item) when name != "Backstage passes to a TAFKAL80ETC concert" do
+    item
+    |> quality_check()
+  end
+
+  defp diff_backstate(item) do
+    item
+    |> reduce_item_quality()
+  end
+
+  defp reduce_item_quality(item) do
+    %{item | quality: item.quality - item.quality}
+  end
+
+
+  defp quality_check(%Item{quality: quality} = item) when quality > 0 do
+    item
+    |> diff_sulfuras_decrease_quality()
+  end
+
+  defp quality_check(item) do
+    item
+  end
+
+  defp reduce_item_sellin(item) do
+    %{item | sell_in: item.sell_in - 1}
+  end
+
+  defp diff_sulfuras_sellin(%Item{name: name} = item) when name != "Sulfuras, Hand of Ragnaros" do
+    item
+    |> reduce_item_sellin()
+  end
+
+  defp diff_sulfuras_sellin(%Item{name: name} = item) do
+    item
+  end
+
+  defp sell_in_less_11(%Item{sell_in: sell_in} = item) when sell_in < 11 do
+    item
+    |> max_quality()
+  end
+
+  defp sell_in_less_11(item) do
+    item
+  end
+
+  defp sell_in_less_6(%Item{sell_in: sell_in} = item) when sell_in < 6 do
+    item
+    |> max_quality()
+  end
+
+  defp sell_in_less_6(item) do
+    item
+  end
+
+  defp diff_sulfuras_decrease_quality(%Item{name: name} = item) when name != "Sulfuras, Hand of Ragnaros" do
+    %{item | quality: item.quality - 1}
+  end
+
+  defp diff_sulfuras_decrease_quality(item) do
+    item
+  end
+
+  defp is_backstage(%Item{name: name} = item) when name == "Backstage passes to a TAFKAL80ETC concert" do
+    item
+    |> sell_in_less_11()
+    |> sell_in_less_6()
+  end
+
+  defp is_backstage(item) do
+    item
+  end
+
+  defp increase_quality_and_backstage(%Item{quality: quality} = item) when quality < 50 do
+    item
+    |> increase_quality()
+    |> is_backstage()
+  end
+
+  defp increase_quality_and_backstage(item) do
+    item
+  end
+
+  defp different_aged_backstage(%Item{name: name} = item) when name != "Aged Brie" and name != "Backstage passes to a TAFKAL80ETC concert" do
+    item
+    |> quality_check()
```

Comment:
idiomatically (And per Credo static-analysis) if you are only piping a single item, it's cleaner or easier to read to just call the function.
```suggestion
  quality_check(item)
```

Reply:
Thank you @ckoch-cars for your feedback! 👍